addons/markers/XEH_postInit.sqf

```diff
@@ -18,3 +18,24 @@ GVAR(currentMarkerPosition) = [];
 GVAR(currentMarkerAngle) = 0;
 GVAR(currentMarkerColorConfigName) = "";
 GVAR(currentMarkerConfigName) = "";
+
+// set marker pos local on every computer (prevent markers visible for everyone)
+[QGVAR(applyMarkerPosLocal), {
+    params["_marker", "_pos"];
+    _marker setMarkerPosLocal _pos;
```

Comment:
Isn't your sample missing the line `GVAR(allMapMarkersProperties) set [_index, _data]` after `_data set [2, _pos];`. The changes I make to `_data` are not set in `GVAR(allMapMarkersProperties)`.